Kernel_23/include/CGAL/Kernel/function_objects.h

```diff
@@ -37,6 +37,39 @@ namespace CGAL {
 
 namespace CommonKernelFunctors {
 
+
+
+  template <typename K>
+  class Non_zero_dimension_3
+  {
+    typedef typename K::Vector_3 Vector_3;
+
+  public:
+    typedef int result_type;
+
+    result_type operator()(const Vector_3& vec) const
+    {
+      if(certainly_not(is_zero(vec.x()))){
+        return 0;
+      } else if(certainly_not(is_zero(vec.y()))){
+        return 1;
+      }else if(certainly_not(is_zero(vec.y()))){
```

Comment:
@afabri Yikes, this is a bug! should be `certainly_not(is_zero(vec.z()))`

Reply:
Thank you for pointing out.